packages/core/src/base/cacheManagerKeyv.ts

```diff
@@ -0,0 +1,230 @@
+    }
+  }
+
+  /**
+   * Returns a Promise that resolves to the response associated with the first
+   * matching request in the Cache object.
+   *
+   * @param req The request to check for a cached response
+   * @param opts Currently for compatibility
+   */
+  async match(req: fetch.Request, opts?: any): Promise<fetch.Response> {
+    const metadataKey: string = getMetadataKey(req);
+    const contentKey: string = getContentKey(req);
+
+    // No match if there's no metadata for this request
+    const redisInfo = await this.keyv.get(metadataKey);
+    if (!redisInfo) {
+      return;
+    }
+
+    // No match if the metadata doesn't match
+    if (
+      !matchDetails(req, {
+        url: redisInfo.metadata.url,
+        reqHeaders: new fetch.Headers(redisInfo.metadata.reqHeaders),
+        resHeaders: new fetch.Headers(redisInfo.metadata.resHeaders),
+        cacheIntegrity: redisInfo.integrity,
+        integrity: opts && opts.integrity
+      })
+    ) {
+      return;
+    }
+
+    // Add customer headers to the response that include caching info
+    const resHeaders: fetch.Headers = new fetch.Headers(
+      redisInfo.metadata.resHeaders
+    );
+    addCacheHeaders(
+      resHeaders,
+      "",
+      contentKey,
+      redisInfo.integrity,
+      redisInfo.time
+    );
+
+    // Return the response without a body for HEAD requests
+    if (req.method === "HEAD") {
+      return new fetch.Response(null, {
+        url: req.url,
+        headers: resHeaders,
+        status: 200
+      });
+    }
+
+    // Get the cached response body
+    const body: string = await this.keyv.get(getContentKey(req));
+
+    return Promise.resolve(
+      new fetch.Response(Buffer.from(body), {
+        url: req.url,
+        headers: resHeaders,
+        status: 200,
+        size: redisInfo.size
+      })
+    );
+  }
+
+  /**
+   * Takes both a request and its response and adds it to the given cache.
+   *
+   * @param req The request this is a storing a response for
+   * @param response The response to be stored
+   * @param opts
+   */
+  async put(
+    req: fetch.Request,
+    response: fetch.Response,
+    opts?: any
+  ): Promise<fetch.Response> {
+    opts = opts || {};
+    const size = response?.headers?.get("content-length");
+    const ckey = getContentKey(req);
+    const cacheOpts = {
+      algorithms: opts.algorithms,
+      integrity: null,
+      metadata: {
+        url: req.url,
+        reqHeaders: req.headers.raw(),
+        resHeaders: response.headers.raw()
+      },
+      size,
+      time: Date.now()
+    };
+    if (req.method === "HEAD" || response.status === 304) {
+      // Update metadata without writing
+      const redisInfo = await this.keyv.get(getMetadataKey(req));
+      // Providing these will bypass content write
+      cacheOpts.integrity = redisInfo.integrity;
+      addCacheHeaders(
+        response.headers,
+        "",
+        ckey,
+        redisInfo.integrity,
+        redisInfo.time
```

Comment:
should cache header update the local cache time with current time on 304 response?

i.e, last param could be a new Date()

Reply:
These are the custom headers and it's meant to indicate the time the response was originally cached. This is actually kept the same as the cacache implementation